app/api/src/utils.py

```diff
@@ -725,3 +726,18 @@ def geopandas_read_file(data_file: UploadFile):
 
         finally:
             delete_file(temp_file_path)
+
+
+async def uniquify_static_layer_name(db, file_name):
+    original_name = file_name.split(".")[0]
+    original_name = tablify(original_name)
+    name = original_name
+    static_layer_table_names = await crud.static_layer.list_static_layer_table_names(
+        db=db, name_like=name
+    )
+    counter = 1
+    while name in static_layer_table_names:
```

Comment:
I was thinking again about this logic and I think it could break in the following scenario.

Imagine you have:

```
table1
table2
table3

```

And now you delete table2. You will have two tables and the counter will be two. Accordingly it will assign 3 for `table3`. But `table3` is already taken and we will run into duplicated table name. Maybe we could get the max counter and then assign max + 1?

Reply:
I think it shouldn't. Because it will be checked again at `while` part.

Reply:
You are right. So we should be fine with this.